src/test/java/redis/clients/jedis/EndpointConfig.java

```diff
@@ -0,0 +1,98 @@
+package redis.clients.jedis;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import redis.clients.jedis.util.JedisURIHelper;
+
+import java.io.FileReader;
+import java.net.URI;
+import java.util.*;
+
+public class EndpointConfig {
+
+    private boolean tls;
+    private String username;
+    private String password;
+    private int bdbId;
+    private Object rawEndpoints;
+    private List<URI> endpoints;
+
+
+    public EndpointConfig(HostAndPort hnp, String username, String password, boolean tls) {
+        this.tls = tls;
+        this.username = username;
+        this.password = password;
+        this.bdbId = 0;
+        this.rawEndpoints = null;
+        this.endpoints = Collections.singletonList(
+                URI.create(getURISchema() + hnp.getHost() + ":" + hnp.getPort())
+        );
+    }
+
+    public HostAndPort getHostAndPort() {
+        return JedisURIHelper.getHostAndPort(endpoints.get(0));
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getHost() {
+        return getHostAndPort().getHost();
+    }
+
+    public int getPort() {
+        return getHostAndPort().getPort();
+    }
+
+    public URI getURI() {
+        return endpoints.get(0);
+    }
+
+    public URI getCustomizedURI(boolean withCredentials, String path) {
+        return getCustomizedURI(withCredentials ? username : "", withCredentials ? password : "", path);
+    }
+
+    public URI getCustomizedURI(String u, String p, String path) {
+        String userInfo = !(u.isEmpty() && p.isEmpty()) ? u + ":" + p + "@" : "";
+        return URI.create(getURISchema() + userInfo + getHost() + ":" + getPort() + path);
+    }
+
+    public Connection getConnection() {
```

Comment:
Being picky here. I would not add to this class the logic of creating Connections or Jedis instances. This will inevitably grow over time. I would only expose the `getClientConfigBuilder` and let each test further customize the builder before calling `build()` on it. If this proves too tedious in the tests, we should fix it in the tests, not here. This class should only be aware of configuration, and not how the configuration can be used.

Reply:
True! We need to fix it in the tests. Removed.